Developer-Admin-UI jar does not work as standalone application anymore

## Problem
Commands to reproduce:
```bash
./gradlew buildDeveloperAdminUI
cp sechub-developertools/build/libs/sechub-developer-admin-ui-0.0.0.jar ~
cd ~
export SECHUB_ADMIN_APITOKEN=my-api-token
export SECHUB_ADMIN_USERID=my-user
export SECHUB_ADMIN_SERVER=localhost
export SECHUB_ADMIN_SERVER_PORT=8443
export SECHUB_ADMIN_ENVIRONMENT=test
java -jar ~/sechub-developer-admin-ui-0.0.0.jar
```
Java run fails with this output:
```
10:57:59.085 [Finalizer] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection manager is shutting down
10:57:59.087 [Finalizer] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection manager shut down
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.mercedesbenz.sechub.integrationtest.internal.MockData.getCombination(MockData.java:77)
        at com.mercedesbenz.sechub.integrationtest.internal.MockData.getTarget(MockData.java:69)
        at com.mercedesbenz.sechub.integrationtest.api.IntegrationTestMockMode.<init>(IntegrationTestMockMode.java:77)
        at com.mercedesbenz.sechub.integrationtest.api.IntegrationTestMockMode.<clinit>(IntegrationTestMockMode.java:20)
        at com.mercedesbenz.sechub.developertools.admin.ui.CommandUI.createIntegrationTestServerMenu(CommandUI.java:440)
        at com.mercedesbenz.sechub.developertools.admin.ui.CommandUI.createMainMenu(CommandUI.java:204)
        at com.mercedesbenz.sechub.developertools.admin.ui.CommandUI.<init>(CommandUI.java:154)
        at com.mercedesbenz.sechub.developertools.admin.ui.DeveloperAdministrationUI.start(DeveloperAdministrationUI.java:90)
        at com.mercedesbenz.sechub.developertools.admin.ui.DeveloperAdministrationUI.main(DeveloperAdministrationUI.java:27)
Caused by: java.lang.IllegalStateException: Testcase corrupt: Cannot read test file /home/myuser/../sechub-other/mockdata/mockdata_setup.json
        at com.mercedesbenz.sechub.test.TestFileReader.loadTextFile(TestFileReader.java:31)
        at com.mercedesbenz.sechub.test.TestFileReader.loadTextFile(TestFileReader.java:12)
        at com.mercedesbenz.sechub.integrationtest.internal.MockedAdapterSetupAccess.<init>(MockedAdapterSetupAccess.java:21)
        at com.mercedesbenz.sechub.integrationtest.internal.MockedAdapterSetupAccess.<clinit>(MockedAdapterSetupAccess.java:16)
        ... 9 more
Caused by: java.io.FileNotFoundException: ../sechub-other/mockdata/mockdata_setup.json (Datei oder Verzeichnis nicht gefunden)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at com.mercedesbenz.sechub.test.TestFileReader.loadTextFile(TestFileReader.java:18)
        ... 12 more
```

--- a/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/CommandUI.java
+++ b/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/CommandUI.java
@@ -32,12 +32,6 @@ import com.mercedesbenz.sechub.developertools.admin.ui.action.config.ListExecuti
 import com.mercedesbenz.sechub.developertools.admin.ui.action.config.ListExecutorConfigurationsAction;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.developerbatchops.DeveloperBatchCreateCheckmarxTestSetupAction;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.FetchMockMailsAction;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.CreateScenario2TestDataAction;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.CreateScenario3TestDataAction;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerMassiveNewJobsScenario3User1Action;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewCodeScanJobScenario3User1Action;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewInfraScanJobScenario3User1Action;
-import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewWebScanJobScenario3User1Action;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.job.CancelJobAction;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.job.DownloadFullscanDataForJobAction;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.job.DownloadHTMLReportForJobAction;
@@ -109,7 +103,6 @@ import com.mercedesbenz.sechub.developertools.admin.ui.action.user.UpdateUserEma
 import com.mercedesbenz.sechub.developertools.admin.ui.action.user.privileges.GrantAdminRightsToUserAction;
 import com.mercedesbenz.sechub.developertools.admin.ui.action.user.privileges.RevokeAdminRightsFromAdminAction;
 import com.mercedesbenz.sechub.domain.scan.product.ProductIdentifier;
-import com.mercedesbenz.sechub.integrationtest.api.IntegrationTestMockMode;
 
 public class CommandUI {
     private static final ImageIcon EDIT_ROAD_BLACK_ICON = new ImageIcon(CommandUI.class.getResource("/icons/material-io/twotone_edit_road_black_18dp.png"));
@@ -418,11 +411,14 @@ public class CommandUI {
     private void createIntegrationTestServerMenu() {
         JMenu menu = new JMenu("Integration TestServer");
         menuBar.add(menu);
+
         if (!ConfigurationSetup.isIntegrationTestServerMenuEnabled()) {
             menu.setEnabled(false);
             menu.setToolTipText("Not enabled, use \"-D" + ConfigurationSetup.SECHUB_ENABLE_INTEGRATION_TESTSERVER_MENU.getSystemPropertyid()
                     + "=true\" to enable it and run an integration test server!");
+            return;
         }
+
         add(menu, new FetchLastStartedPDSJobStreamsAction(context));
         add(menu, new FetchMockMailsAction(context));
         menu.addSeparator();
@@ -432,23 +428,15 @@ public class CommandUI {
 
         JMenu testDataMenu = new JMenu("Testdata");
         menu.add(testDataMenu);
-        add(testDataMenu, new CreateScenario2TestDataAction(context));
-        add(testDataMenu, new CreateScenario3TestDataAction(context));
-        testDataMenu.addSeparator();
-        add(testDataMenu, new TriggerNewInfraScanJobScenario3User1Action(context));
-        testDataMenu.addSeparator();
-        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__ZERO_WAIT));
-        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__10_SECONDS_WAITING));
 
-        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__10_SECONDS_WAITING));
+        try {
+            new IntegrationTestDataMenuAppender().appendMenuEntries(context, testDataMenu);
+        } catch (ExceptionInInitializerError e) {
 
-        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_RED__ZERO_WAIT));
-        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_MULTI__ZERO_WAIT));
-        testDataMenu.addSeparator();
-        add(testDataMenu, new TriggerNewCodeScanJobScenario3User1Action(context, IntegrationTestMockMode.CODE_SCAN__CHECKMARX__MULTI__ZERO_WAIT));
-        add(testDataMenu, new TriggerNewCodeScanJobScenario3User1Action(context, IntegrationTestMockMode.CODE_SCAN__CHECKMARX__GREEN__1_SECOND_WAITING));
-        testDataMenu.addSeparator();
-        add(testDataMenu, new TriggerMassiveNewJobsScenario3User1Action(context));
+            testDataMenu.add(ActionSupport.getInstance().createErrorReplacementAction("Cannot add test data actions",
+                    "Was not able to initialize test data actions.\n" + "This works only from IDE and not from standalone application.\n" + "\n"
+                            + "Explanation: The necessary mock data file is not published but only available inside sources."));
+        }
 
     }
 

--- a/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/IntegrationTestDataMenuAppender.java
+++ b/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/IntegrationTestDataMenuAppender.java
@@ -1,0 +1,53 @@
+package com.mercedesbenz.sechub.developertools.admin.ui;
+
+import javax.swing.AbstractAction;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.CreateScenario2TestDataAction;
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.CreateScenario3TestDataAction;
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerMassiveNewJobsScenario3User1Action;
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewCodeScanJobScenario3User1Action;
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewInfraScanJobScenario3User1Action;
+import com.mercedesbenz.sechub.developertools.admin.ui.action.integrationtestserver.testdata.TriggerNewWebScanJobScenario3User1Action;
+import com.mercedesbenz.sechub.integrationtest.api.IntegrationTestMockMode;
+
+public class IntegrationTestDataMenuAppender {
+
+    public void appendMenuEntries(UIContext context, JMenu testDataMenu) {
+        assertMockDataAvailable();
+
+        add(testDataMenu, new CreateScenario2TestDataAction(context));
+        add(testDataMenu, new CreateScenario3TestDataAction(context));
+        testDataMenu.addSeparator();
+        add(testDataMenu, new TriggerNewInfraScanJobScenario3User1Action(context));
+        testDataMenu.addSeparator();
+        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__ZERO_WAIT));
+        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__10_SECONDS_WAITING));
+
+        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_GREEN__10_SECONDS_WAITING));
+
+        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_RED__ZERO_WAIT));
+        add(testDataMenu, new TriggerNewWebScanJobScenario3User1Action(context, IntegrationTestMockMode.WEBSCAN__NETSPARKER_MULTI__ZERO_WAIT));
+        testDataMenu.addSeparator();
+        add(testDataMenu, new TriggerNewCodeScanJobScenario3User1Action(context, IntegrationTestMockMode.CODE_SCAN__CHECKMARX__MULTI__ZERO_WAIT));
+        add(testDataMenu, new TriggerNewCodeScanJobScenario3User1Action(context, IntegrationTestMockMode.CODE_SCAN__CHECKMARX__GREEN__1_SECOND_WAITING));
+        testDataMenu.addSeparator();
+        add(testDataMenu, new TriggerMassiveNewJobsScenario3User1Action(context));
+    }
+
+    private void assertMockDataAvailable() {
+        /*
+         * triggering next method, ensures that we have access to mock data - if not,
+         * there will be an exception (which is caught at caller side)
+         */
+        IntegrationTestMockMode.values();
+    }
+
+    private void add(JMenu menu, AbstractAction action) {
+        JMenuItem menuItem = new JMenuItem(action);
+        menu.add(menuItem);
+
+    }
+
+}

--- a/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/action/ActionSupport.java
+++ b/sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/admin/ui/action/ActionSupport.java
@@ -13,6 +13,7 @@ import javax.swing.ActionMap;
 import javax.swing.InputMap;
 import javax.swing.JMenu;
 import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
 import javax.swing.JPopupMenu;
 import javax.swing.KeyStroke;
 import javax.swing.text.DefaultEditorKit;
@@ -127,6 +128,29 @@ public class ActionSupport {
         List<JMenuItem> menuItems = new ArrayList<>();
         menuItems.add(new JMenuItem(new CleanOutputAreaAction(component)));
         return menuItems;
+    }
+
+    public Action createErrorReplacementAction(String text, String description) {
+        return new ErrorReplacementAction(text, description);
+    }
+
+    private class ErrorReplacementAction extends AbstractAction {
+
+        private static final long serialVersionUID = 1L;
+        private String description;
+        private String title;
+
+        private ErrorReplacementAction(String title, String description) {
+            super(title);
+            this.title = title;
+            this.description = description;
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            JOptionPane.showMessageDialog(null, description, title, JOptionPane.ERROR_MESSAGE);
+        }
+
     }
 
     private class UndoAction extends AbstractAction {

